pkg/unpack/reflector.go

```diff
@@ -0,0 +1,404 @@
+package unpack
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"github.com/mitchellh/mapstructure"
+)
+
+var zero reflect.Value
+
+type Reflector map[string]map[string]reflect.Type
+
+func New() Reflector {
+	return make(map[string]map[string]reflect.Type)
+}
+
+func (r Reflector) Init(templates ...interface{}) Reflector {
+	for _, t := range templates {
+		r.Add(t)
+	}
+	return r
+}
```

Comment:
Nit: I'd get rid of `Init` and allow `New` to accept templates.
```suggestion
func New(templates ...interface{}) Reflector {
	r := Reflector(make(map[string]map[string]reflect.Type))
	for _, t := range templates {
		r.Add(t)
	}
	return r
}
```